scripts/add.tokens.erc20.mjs

```diff
@@ -0,0 +1,377 @@
+		newFileContentMainnet +
+		newFileContentTestnet;
+	writeFileSync(filePath, newFileContent);
+	filesCreatedOrModified = true;
+
+	if (newFileContentMainnet !== '') {
+		console.log(`Created new mainnet variables for token ${symbol} in file ${filePath}`);
+	}
+
+	if (newFileContentTestnet !== '') {
+		console.log(`Created new testnet variables for token ${symbol} in file ${filePath}`);
+	}
+
+	return {
+		fileName,
+		mainnetToken: newFileContentMainnet !== '' ? mainnetToken : undefined,
+		testnetToken: newFileContentTestnet !== '' ? testnetToken : undefined
+	};
+};
+
+const updateTokensErc20Env = ({ fileName, mainnetToken, testnetToken }) => {
+	const filePath = path.join(DATA_DIR_PATH, 'tokens.erc20.env.ts');
+	let content = readFileSync(filePath, 'utf8');
+
+	const regexList = [
+		{
+			regex: /const ERC20_TWIN_TOKENS_MAINNET: RequiredErc20Token\[] = \[([\s\S]*?)];/,
+			token: mainnetToken
+		},
+		{
+			regex: /const ERC20_TWIN_TOKENS_SEPOLIA: RequiredErc20Token\[] = \[([\s\S]*?)];/,
+			token: testnetToken
+		}
+	];
+
+	regexList
+		.filter(({ token }) => nonNullish(token))
+		.forEach(({ regex, token }) => {
+			const { content: newContent, hasChanged } = addElementListInContent({
+				content,
+				regex,
+				element: token
+			});
+			if (hasChanged) {
+				content = newContent;
+				console.log(`Included ${token} in token list in ${filePath}`);
+			}
+		});
+
+	if (nonNullish(mainnetToken) || nonNullish(testnetToken)) {
+		const imports = [mainnetToken, testnetToken].filter(Boolean);
+		const { content: newContent, hasChanged } = updateImportsInContent({
+			content,
+			imports,
+			module: `$env/${fileName.replace('.ts', '')}`
+		});
+		if (hasChanged) {
+			content = newContent;
+			console.log(`Added import statement for ${imports.join(', ')} in ${filePath}`);
+		}
+	}
+
+	writeFileSync(filePath, content);
+};
+
+const flattenData = (data) => {
+	return Object.keys(data).map((symbol) => ({
+		symbol,
+		...data[symbol]
+	}));
+};
+
+const flattenEnvironmentData = (data) => {
+	return Object.entries(data).reduce(
+		(acc, [environment, values]) => ({
+			...acc,
+			[environment]: flattenData(values)
+		}),
+		{}
+	);
+};
+
+const readSupportedTokens = () => {
+	const jsonPath = path.resolve(DATA_DIR_PATH, 'tokens.ckerc20.json');
+	const { production: prodTokens, staging: testnetTokens } = flattenEnvironmentData(
+		JSON.parse(readFileSync(jsonPath, 'utf-8'))
+	);
+	return [...prodTokens, ...testnetTokens];
+};
+
+const parseTokens = (tokens) =>
+	Object.values(
+		tokens.reduce((acc, { symbol, erc20ContractAddress }) => {
+			const mainSymbol = symbol.replace('Sepolia', '').slice(2);
+			const contractAddress = !symbol.includes('ckSepolia') ? erc20ContractAddress : undefined;
+			const testnetContractAddress = symbol.includes('ckSepolia')
+				? erc20ContractAddress
+				: undefined;
+
+			acc[mainSymbol] = {
+				mainSymbol,
+				contractAddress: acc[mainSymbol]?.contractAddress || contractAddress,
```

Comment:
The linter still does not catch nullish coalish `??` instead of `||`?

Reply:
Strangely enough no! I am not really sure what happened, because a few times, it caught it. Why not this time I am not sure. I will investigate

Reply:
As discussed, this is solved